src/testflinger_device_connectors/devices/oemrecovery/oemrecovery.py

```diff
@@ -89,6 +91,68 @@ def provision(self):
         self._run_cmd_list(recovery_cmds)
         self.check_device_booted()
 
+    @timeout(dec_timeout=3600)
+    def check_boot_state(self):
+        """check boot state"""
+        try:
+            test_username = self.job_data.get("test_data", {}).get(
+                "test_username", "ubuntu"
+            )
+            test_password = self.job_data.get("test_data", {}).get(
+                "test_password", "ubuntu"
+            )
+        except AttributeError:
+            test_username = "ubuntu"
+            test_password = "ubuntu"
+
+        while True:
+            try:
+                image_type = (
+                    subprocess.check_output(
+                        (
+                            "sshpass -p {} ssh -o StrictHostKeyChecking=no "
+                            "-o UserKnownHostsFile=/dev/null "
+                            "{}@{} snap model | grep model"
+                        ).format(
+                            test_password,
+                            test_username,
+                            self.config["device_ip"],
+                        ),
+                        shell=True,
+                    )
+                ).decode()
+            except Exception:
+                time.sleep(10)
+                continue
+
+            if image_type.find("generic-classic") != -1:
```

Comment:
I don't think we can trust that generic-classic will be the type. oemrecovery is also used with things like core20/22 FDE systems. For example, one device here that I have has this output for `snap model`:
```
$ snap model
brand   Canonical✓
model   ubuntu-core-20-amd64-dangerous
grade   dangerous
...
```

Reply:
Hi Paul
the  "generic-classic" is for classic image and it will do nothing but return directly. We only check "Initialize Device" on core image